spec/unit/provider/user/useradd_spec.rb

```diff
@@ -318,24 +318,48 @@
   end
 
   describe "#comment" do
-    before do
-      described_class.has_feature :libuser
-    end
+    before { described_class.has_feature :libuser }
 
-    let(:passwd_line) { "myuser:x:1492:1000:local comment:/home/myuser:/bin/bash" }
+    let(:user_object) { { gecos: 'local comment' } }
 
     it "should return the local comment string when forcelocal is true" do
       resource[:forcelocal] = true
-      allow(provider).to receive(:finduser).with('account', 'myuser').and_return(passwd_line)
-      expect(provider).to receive(:localcomment).and_return('local comment')
-      provider.comment
+      allow(provider).to receive(:finduser).with(:account, 'myuser').and_return(user_object)
```

Comment:
stubbing the `finduser` method is a code smell since it's part of the code that we're trying to test. It would be better to stub what `File.open` returns.

Reply:
I thought it'd be cleaner to stub `finduser` here since I'm already testing it separately below, and its "usage" is not really in the scope of the `comment` method which I'm testing here. Is this the way to go?

Reply:
Makes sense, I changed it.